### DISCUSSION TITLE
Deploy in Android App

### DISCUSSION THREAD

**Discussion by evanescenceakii** [microsoft/onnxruntime#7637] [General]
I'm trying to create an Android App that incorporates a Machine Learning Model. I had an onnx model, along with a Python script file, two json files with the label names, and some numpy data for mel spectrograms computation.

I tried to go with onnxruntime, and followed [these ](https://github.com/microsoft/onnxruntime/blob/master/docs/ONNX_Runtime_for_Mobile_Platforms.md)instructions. So now I have created the model.ort file out of the onnx model and "A minimal build for Android with NNAPI support", so I have the Build onnxruntime pkg. I'm run the build in windows cmd.

Since I'm completely new at this, how do I continue from here? How do I "inference on device" and deploy in my phone?

And also, will I have to convert my python script that runs the model to Java? Or is there any other way? Thank you!

**Comment by guoyu-wang** [comment]
Do you have an Android application ready for the integration of Onnx Runtime for mobile inference? 
Since the actual integration of Onnx Runtime Mobile is highly dependable on your actual application and the language you use, you will have to convert the python script to either c/c++ or java/kotlin.

To use Onnx Runtime c/c++ API,
- See examples of using c/c++ API here, https://github.com/microsoft/onnxruntime/tree/master/samples/c_cxx
- Also you may want to check the [Android NDK samples](https://github.com/android/ndk-samples) on how to integrate native libraries in your Android App, to be more specific, look at this particular sample app, https://github.com/android/ndk-samples/tree/main/hello-libs
- Here is an example app running MobileNet v2 using Onnx Runtime c++ API, https://github.com/Rohithkvsp/OnnxRuntimeAndorid

To use Onnx Runtime java/kotlin API,
- See examples of using java API here, https://github.com/microsoft/onnxruntime/tree/master/java/src/test/java/sample
- Also you may want to check this sample Android App here, https://github.com/gwang-msft/ort_image_classification_android, which integrates MobileNet v1 in an Android image classification App using kotlin
